src/Map/Sketch/hooks.ts

```diff
@@ -69,21 +74,33 @@ export default function useHooks({
   onSketchTypeChange,
   onSketchFeatureCreate,
   onSketchPluginFeatureCreate,
+  onSketchFeatureUpdate,
+  onSketchPluginFeatureUpdate,
   onLayerSelect,
+  sketchEditingFeature,
+  onSketchEditFeature,
+  onMount,
 }: Props) {
   const [state, send] = useMachine(sketchMachine);
   const [type, updateType] = useState<SketchType | undefined>();
   const [from, updateFrom] = useState<"editor" | "plugin">("editor");
```

Comment:
BTW what's the use case of `from`?

Reply:
Emmm it's an problem left over by history. (coming from the original implement when core is still within Visualizer).
Basically that's for different behaviors that triggers from Editor or Plugin API. Now it should be handled on Visualizer side, but there's more structure related need to refactor. So i want to refactor them together later.